### PR TITLE
client/login: add login method

### PR DESCRIPTION
**Pull request by matiasb**
Add login method for daemon REST login API.

### CODE REVIEW

File: client/login.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+type AuthenticatedUser struct {
+	Username   string   `json:"username,omitempty"`
+	Macaroon   string   `json:"macaroon,omitempty"`
+	Discharges []string `json:"discharges,omitempty"`
+}
+
+// Login logs user in.
+func (client *Client) Login(username, password, otp string) (*AuthenticatedUser, error) {
+	var user AuthenticatedUser
+	body := strings.NewReader(fmt.Sprintf(`{"username":%q,"password":%q,"otp":%q}`, username, password, otp))
```
Comment (niemeyer) [inline client/login.go]: Can we please use json itself to marshal this out?  I know this technique was used elsewhere, but it feels a bit dubious, even more when dealing with open-ended auth data. You can easily do that with something like:
```
var body bytes.Buffer
err := json.NewEncoder(&buf).Encode(value)
```
Then use `&body` directly as the body parameter.

File: client/login.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	err := client.doSync("POST", "/v2/login", nil, body, &user)
+	if err != nil {
+		return nil, err
+	}
+	if err := writeAuthData(user); err != nil {
+		return nil, fmt.Errorf("cannot store login information: %v", err)
```
Comment (niemeyer) [inline client/login.go]: s/store/persist/, to avoid the ambiguity with that other store. :-)

File: client/login.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	return &user, err
+}
+
+func storeAuthDataFilename() string {
+	homeDir, _ := osutil.CurrentHomeDir()
```
Comment (niemeyer) [inline client/login.go]: What is that second argument? An error?

If you expect it to never be != nil, let's please check and panic properly if it is. Silently ignored errors is great source of very curious bugs.

File: client/login.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	return filepath.Join(homeDir, ".snap", "auth.json")
+}
+
+func writeAuthData(userData AuthenticatedUser) error {
+	targetFile := storeAuthDataFilename()
+	if err := os.MkdirAll(filepath.Dir(targetFile), 0750); err != nil {
```
Comment (niemeyer) [inline client/login.go]: 0700?

**Comment by niemeyer** [comment]
LGTM

File: client/login_test.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+)
+
+func (cs *clientSuite) TestClientLogin(c *check.C) {
+	cs.rsp = `{"type": "sync", "result":
+                     {"username": "the-user-name",
+                      "macaroon": "the-root-macaroon",
+                      "discharges": ["discharge-macaroon"]}}`
+
+	tmpdir := c.MkDir()
+	os.Setenv("HOME", tmpdir)
```
Comment (chipaca) [inline client/login_test.go]: could you set it back to the original value when you're done?

**Comment by chipaca** [comment]
:+1:

**Pull request merged**